src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -1046,12 +1046,16 @@ protected void GetFeatureIniContent(RoleMappedData data, ref VBuffer<ReadOnlyMem
         {
             Contracts.Assert(!values.IsDense);
             Contracts.Assert(Algorithms.FindFirstGE(binUpperBounds, 0) == 0);
-            for (int i = 0; i < values.Count; ++i)
+            var valuesValues = values.GetValues();
+            var valuesIndices = values.GetIndices();
+            List<KeyValuePair<int, int>> result = new List<KeyValuePair<int, int>>();
```

Comment:
>List<KeyValuePair<int, int>> [](start = 12, length = 28)

nit: var? 
it just stand out:)

Reply:
I'll fix it the `var` here to be the real type.

> also any reason why you prefer return IEnumerable to yield?

It is required because of the Span usage. You can't use a Span in a `yield` method because you can't hold onto a Span across the `yield` invocations.